lib/zabbix/zabbix_test.go

```diff
@@ -0,0 +1,98 @@
+package zabbix
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+)
+
+// Test that we can get Zabbix API key from an API call,
+// set it in API struct
+func TestGetAndSetKey(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		fmt.Fprintln(w, `{"result": "fake_zabbix_key"}`)
+	}))
+	defer ts.Close()
+
+	var api API
+	api.User = "ZABBIX_USER"
+	api.Password = "ZABBIX_PASSWORD"
+	api.URL = ts.URL
+
+	// Get API Key
+	key, err := api.getKey()
+	if err != nil {
+		t.Fatalf("Process ran with err %v, want ZABBIX_API_KEY to be fake_zabbix_key", err)
+		return
+	}
+	if key != "fake_zabbix_key" {
+		t.Error("Expected key to be fake_zabbix_key")
+	}
+
+	// Set API Key
+	api.SetKey()
+
+	if api.Key != "fake_zabbix_key" {
+		t.Error("Expected ZabbixAPI.Key to be fake_zabbix_key")
+	}
+}
+
+// Test that given a payload,
+// we receive a response when we make an API call
+func TestMakeRequest(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+	}))
+	defer ts.Close()
+
+	var api API
+	api.URL = ts.URL
+
+	payload := api.BuildPayload(nil, "apiinfo.version")
+	resp, err := api.MakeRequest(payload)
+	if err != nil {
+		t.Fatalf("Process ran with err %v, want response", err)
+		return
```

Comment:
When `t.Fatalf` is invoked, this line is never reached.
This is because of tests exit.